staging/src/k8s.io/apiserver/pkg/audit/policy/reader.go

```diff
@@ -61,11 +62,22 @@ func LoadPolicyFromFile(filePath string) (*auditinternal.Policy, error) {
 
 func LoadPolicyFromBytes(policyDef []byte) (*auditinternal.Policy, error) {
 	policy := &auditinternal.Policy{}
+
+	strictDecoder := serializer.NewCodecFactory(audit.Scheme, serializer.EnableStrict).UniversalDecoder(apiGroupVersions...)
 	decoder := audit.Codecs.UniversalDecoder(apiGroupVersions...)
 
-	_, gvk, err := decoder.Decode(policyDef, nil, policy)
+	// Try strict decoding first.
+	_, gvk, err := strictDecoder.Decode(policyDef, nil, policy)
 	if err != nil {
-		return nil, fmt.Errorf("failed decoding: %v", err)
+		if !runtime.IsStrictDecodingError(err) {
+			return nil, fmt.Errorf("failed decoding: %v", err)
+		}
+		var lenientErr error
+		_, gvk, lenientErr = decoder.Decode(policyDef, nil, policy)
+		if lenientErr != nil {
+			return nil, lenientErr
+		}
+		klog.Warningf("Using lenient decoding as strict decoding failed, %v", err)
```

Comment:
Let's make this error a bit more actionable to the user
```suggestion
		klog.Warningf("Audit policy contains errors, falling back to lenient decoding: %v", err)
```